Lib/multiprocessing/pool.py

```diff
@@ -429,15 +440,28 @@ def _map_async(self, func, iterable, mapper, chunksize=None, callback=None,
         )
         return result
 
+    def _wait_for_updates(self, timeout):
+        task_queue_sentinels = [self._outqueue._reader.fileno(),
+                                self._outqueue._writer.fileno()]
+        self_notifier_sentinels = [self._change_notifier._reader.fileno()]
+        worker_sentinels = [worker.sentinel for worker in self._pool]
+        sentinels = [*task_queue_sentinels,
+                     *worker_sentinels,
+                     *self_notifier_sentinels]
+        wait(sentinels, timeout=timeout)
```

Comment:
Please add a comment to explain why you kept a timeout here.

I dislike keeping the timeout ("active" polling). I would prefer to experiment without timeout. If something hangs: try to understand why and maybe try to fix it. I would only add back the timeout if there is a design issue that we cannot work around or fix.

Reply:
If you remove the timeout, everything works as expected, but I prefer to maintain the timeout here for backwards compatibility, so if we forgot something, there is not a deadlock or a hang. The pool has a lot of moving pieces and even if we think we have everything covered we could miss something, so I feel safer to leave the fallback.

The expected behaviour is that the timeout will never be really needed and the underlying select in wait will wake up earlier. Of course, if there are no new jobs/changes/...etc the timeout will be hit but it will do nothing.

What do you think? Do you agree or you really want to delete the timeout? :)

Reply:
Let's remove the timeout (or make it very long, e.g. 10 seconds). This is only going in 3.8 in any case.